lib/ansible/modules/assemble.py

```diff
@@ -265,13 +266,16 @@ def main():
                 module.fail_json(msg="failed to validate: rc:%s error:%s" % (rc, err))
         if backup and dest_hash is not None:
             result['backup_file'] = module.backup_local(dest)
+            if module.check_mode:
+                os.unlink(result['backup_file'])
```

Comment:
Could you add check mode support to `backup_local` instead, so it just returns the backup file, and avoid the need to delete it here? Otherwise, lgtm.